### PR TITLE
Migrate bicep example: 101/databricks-all-in-one-template-for-vnet-injection

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Added bicep support to this sample by integrating bicep.main from the example in https://github.com/Azure/bicep/tree/main/docs/examples/101/databricks-all-in-one-template-for-vnet-injection

https://github.com/Azure/bicep/pull/3151

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
fyi @sunainakrishnamoorthy you're welcome to review if you'd like.

File: quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+@description('The name of the Azure Databricks workspace to create.')
+param workspaceName string
+
+var managedResourceGroupName = 'databricks-rg-${workspaceName}-${uniqueString(workspaceName, resourceGroup().id)}'
+
+resource nsg 'Microsoft.Network/networkSecurityGroups@2020-05-01' = {
+  location: location
+  name: nsgName
+  properties: {
+    securityRules: [
```
Comment (mvvsubbu) [inline quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:45]: Thanks Stephen for removing excess variables in ARM template and making it more cleaner. We also don't NSG rules, can you remove them as well in ARM template and Bicep file as well?

File: quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+@allowed([
+  'trial'
+  'standard'
+  'premium'
+])
+@description('The pricing tier of workspace.')
+param pricingTier string = 'premium'
+
+@description('Cidr range for the private subnet.')
```
[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:18]: Did for description but not var names (cause camel case)

File: quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep (hunk 1 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:45]: @mvvsubbu  Could you be more specific on what I should remove?  Note that everything in here was in the original sample.  thx

File: quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      }
+      customPrivateSubnetName: {
+        value: privateSubnetName
+      }
+      enableNoPublicIp: {
+        value: disablePublicIp
+      }
+    }
+  }
+  dependsOn: [
```
[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:205]: The decompiler put it in there.  workspace -> vnet -> nsg.

It's redundant, so I'll remove it.

[the remark this answers is not in the document]
Comment (bmoore-msft) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:205]: I'm not seeing the changes - did you push yet?

[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep]: Sorry, now I have.

File: quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep (hunk 1 above)
Comment (mvvsubbu) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:50]: @StephenWeatherford I am asking help to remove securityRules in NSG resources as we don't need them.

Comment (StephenWeatherford) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:50]: The entire securityRules property?  I don't have any knowledge of the workings of this sample, I've only converted it.

@bmoore-msft can you help?

Comment (bmoore-msft) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:50]: Let's separate the two issues - this PR is just about bicep conversion, not "clean-up".  If we also want to clean up the sample, let's do a separate PR from a domain expert...

Comment (mvvsubbu) [reply quickstarts/microsoft.databricks/databricks-all-in-one-template-for-vnet-injection/main.bicep:50]: Ok sure.

**Pull request merged**